### PR TITLE
Jetpack Connect: Use withTrackingTool for HotJar

### PR DESCRIPTION
**Pull request by tyxla**
This PR refactors Jetpack Connect to use `withTrackingTool` high-order component for loading HotJar - we introduced it in #30809. Previously, we were manually dispatching the load action in every component of interest.

#### Changes proposed in this Pull Request

* Refactor Jetpack Connect to use `withTrackingTool` high-order component for loading HotJar.

#### Testing instructions

* Checkout this branch.
* Run this in your browser console: `localStorage.setItem( 'debug', 'calypso:analytics:hotjar' )`
* Go through all affected Jetpack Connect flows.
* Verify `hotjar` still gets loaded on each of the pages by seeing `calypso:analytics:hotjar Not loading HotJar script` in the console on every first load of each page.

Pinging @Automattic/jetpack-design in case they're interested to help test this.

### CODE REVIEW

File: client/jetpack-connect/authorize.js [hunk 1]
```diff
@@ -692,7 +689,7 @@ export class JetpackAuthorize extends Component {
 	}
 }
 
-export default connect(
+const connectComponent = connect(
```
Comment (ockham) [inline client/jetpack-connect/authorize.js]: Bit of a nitpick: `connectComponent` sounds a bit like a verb/action, so I think using the participle  (`connectedComponent`) would be more descriptive in cases like this.

Comment (sirreal) [reply client/jetpack-connect/authorize.js]: It is a verb 🙂 

It's "partially applied," connect is `connect( mapStateToProps, mapDispatchToProps )( WrappedComponent )`. Here it is a function ready to receive a component to connect.

Comment (ockham) [reply client/jetpack-connect/authorize.js]: :man_facepalming: You're obviously right @sirreal.

Ignore me please :grimacing:

Comment (tyxla) [reply client/jetpack-connect/authorize.js]: Yup, what Jon said 😉 Our `connect` calls here return a function, and calling _that_ function connects the component. Which essentially justifies the verb.

Comment (tyxla) [reply client/jetpack-connect/authorize.js]: > Ignore me please 😬

I never do 😛

**Comment by ockham** [comment]
Code looks good. I've tried two JP Connect flows:

- Starting from `calypso.localhost:3000/jetpack/connect`, I get `calypso:analytics:hotjar Not loading HotJar script` :white_check_mark: 
- Starting from a newly created JN site's wp-admin (with `&calypso_env` appended to the connect link), I can't seem to get that message.

**Comment by tyxla** [comment]
> * Starting from a newly created JN site's wp-admin (with `&calypso_env` appended to the connect link), I can't seem to get that message.

You'll only get it when `localStorage.setItem( 'debug', 'calypso:analytics:hotjar' )` has been ran in that session. I just tried and it worked for me. Could you verify you ran this snippet locally before trying it out from wp-admin?

**Comment by ockham** [comment]
>> * Starting from a newly created JN site's wp-admin (with `&calypso_env` appended to the connect link), I can't seem to get that message.
>
> You'll only get it when `localStorage.setItem( 'debug', 'calypso:analytics:hotjar' )` has been ran in that session. I just tried and it worked for me. Could you verify you ran this snippet locally before trying it out from wp-admin?

I did:

1. Incognito window
2. Create new JN
3. In a different tab: `calypso.localhost:3000/jetpack/connect`, browser console, enter `localStorage.setItem( 'debug', 'calypso:analytics:hotjar' )`
4. Copy JP Connect link from JN, append `&calypso_env=development`
5. No `calypso:analytics:hotjar Not loading HotJar script` :man_shrugging: 

I think I've managed to repro the `calypso:analytics:hotjar Not loading HotJar script` thing once by reloading during JPC, but it's been spurious...

**Comment by tyxla** [comment]
@ockham are you landing on the login page? I think this is expected then - we don't add HotJar for the login page currently AFAIK.

Could you please compare current production behavior vs this branch? This branch doesn't introduce any new HotJar usages, it only migrates the already existing ones. 

Thanks!

**Comment by tyxla** [comment]
Thanks, folks!

**Pull request merged**